_core_upgrader.py

```diff
@@ -253,25 +252,14 @@ def upgrade_tank(sgtk_install_root, log):
             backup_folder_name = datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
             backup_path = os.path.join(core_backup_location, backup_folder_name)
             log.info("Backing up Core API: %s -> %s" % (core_install_location, backup_path))
-            
-            # first copy the content in the core folder
-            src_files = _copy_folder(log, core_install_location, backup_path)
-            
-            # now clear out the install location
-            log.info("Clearing out target location...")
-            for f in src_files:
-                try:
-                    # on windows, ensure all files are writable
-                    if sys.platform == "win32":
-                        attr = os.stat(f)[0]
-                        if (not attr & stat.S_IWRITE):
-                            # file is readonly! - turn off this attribute
-                            os.chmod(f, stat.S_IWRITE)
-                    os.remove(f)
-                    log.debug("Deleted %s" % f)
-                except Exception, e:
-                    log.error("Could not delete file %s: %s" % (f, e))
-            
+
+            # move the folder to the backup location
+            try:
+                shutil.move(core_install_location, backup_path)
```

Comment:
The current approach is to handle the story where a core upgrade for some reason fails halfway. With the old implementation, you would have two scenarios:

- The copy phase fails for some reason. The upgrade therefore fails but the current core install is left intact and working.
- The copy phase works, a backup is succesfully made but the delete fails. User can simply move the backup into position to unbreak.
- The copy phase works, the delete phase works but the install phase fails. User can simply move the backup into position to unbreak.

Is there some way to retain a similar integrity with the `shutil.move` approach? We basically don't want to get stuck half way without a way for a user to successfully revert. Is the move operation atomic enough to provide that level of confidence? On a local file system, i think it's ok - but on a network storage, i think it does a copy/then delete type thing - what happens if this fails half way?